### PR TITLE
Add surveys

### PR DESCRIPTION
**Pull request by bqqbarbhg**
Add support for creating and answering surveys

### CODE REVIEW

File: client/src/components/open-survey-panel-button/open-survey-panel-button.js
Comment (HenrikHelenius) [inline client/src/components/open-survey-panel-button/open-survey-panel-button.js]: use syntax: {'hidden': hidden}

File: client/src/components/survey-edit/survey-edit-card.js
Comment (HenrikHelenius) [inline client/src/components/survey-edit/survey-edit-card.js]: quite complicated tertiary clause, is it still fine? your call

File: client/src/components/survey-edit/survey-edit-container.js
Comment (HenrikHelenius) [inline client/src/components/survey-edit/survey-edit-container.js]: break lines in buttons

File: client/src/components/survey-edit/survey-edit-container.scss
Comment (HenrikHelenius) [inline client/src/components/survey-edit/survey-edit-container.scss:52]: are these duplicated styles somewhere, consider extending @joonalillfors

File: client/src/components/survey/answer.js
Comment (HenrikHelenius) [inline client/src/components/survey/answer.js]: newline after {

File: client/src/components/survey/answer.js
Comment (HenrikHelenius) [inline client/src/components/survey/answer.js]: } to own line

File: client/src/components/survey/answer.js
Comment (HenrikHelenius) [inline client/src/components/survey/answer.js]: { and } fix

File: client/src/components/survey/survey.js
Comment (HenrikHelenius) [inline client/src/components/survey/survey.js]: todos?

File: client/src/components/survey/survey.js
Comment (HenrikHelenius) [inline client/src/components/survey/survey.js]: can be removed as we disable the button if ther is not state

File: client/src/components/survey/survey.js
Comment (HenrikHelenius) [inline client/src/components/survey/survey.js]: To variable as it will be used in tooltip also

**Review by HenrikHelenius — CHANGES_REQUESTED** [review]

File: client/src/components/survey-edit/survey-edit-card.js
Comment (tobbe-j) [reply client/src/components/survey-edit/survey-edit-card.js]: Could these be moved to functions to make code a bit clearer?

File: server/src/database.js
Comment (tobbe-j) [inline server/src/database.js:82]: This could be added as a view to the database to simplify (and optimize?) so this query could be:
SELECT * FROM question_with_answer_view

File: server/src/database/migrations/sqlite/001-setup.sql
Comment (tobbe-j) [inline server/src/database/migrations/sqlite/001-setup.sql]: quite dangerous to delete database values of container if user gets deleted, will lead to unknown instances continuing running without the program knowing?

Comment (bqqbarbhg) [reply server/src/database/migrations/sqlite/001-setup.sql]: Yea that is not good, should not do this then I guess

File: client/src/scss/views/dashboard-view.scss
Comment (bqqbarbhg) [inline client/src/scss/views/dashboard-view.scss]: Committed diff markers?

File: client/src/components/survey/survey.js
Comment (joonalillfors) [reply client/src/components/survey/survey.js]: can not be removed, if it's removed the app will crash as the survey panel is still mounted, just not displayed

File: client/src/scss/views/dashboard-view.scss
Comment (joonalillfors) [reply client/src/scss/views/dashboard-view.scss]: solved

File: client/src/components/survey/survey.js
Comment (joonalillfors) [reply client/src/components/survey/survey.js]: solved

File: client/src/components/survey/answer.js
Comment (joonalillfors) [reply client/src/components/survey/answer.js]: solved

File: client/src/components/survey/answer.js
Comment (joonalillfors) [reply client/src/components/survey/answer.js]: solved

File: client/src/components/survey/answer.js
Comment (joonalillfors) [reply client/src/components/survey/answer.js]: solved

File: client/src/components/survey-edit/survey-edit-container.js
Comment (joonalillfors) [reply client/src/components/survey-edit/survey-edit-container.js]: solved

File: client/src/components/open-survey-panel-button/open-survey-panel-button.js
Comment (joonalillfors) [reply client/src/components/open-survey-panel-button/open-survey-panel-button.js]: donezo

File: client/src/components/survey-edit/survey-edit-container.scss
Comment (joonalillfors) [reply client/src/components/survey-edit/survey-edit-container.scss:52]: this fine

Comment (joonalillfors) [reply client/src/components/survey-edit/survey-edit-container.scss:52]: no duplications

File: client/src/components/survey/survey.js
Comment (joonalillfors) [reply client/src/components/survey/survey.js]: donezo

File: client/src/components/survey-edit/survey-edit-card.js
Comment (joonalillfors) [reply client/src/components/survey-edit/survey-edit-card.js]: solved

Comment (joonalillfors) [reply client/src/components/survey-edit/survey-edit-card.js]: do nothing, ship it

**Review by HenrikHelenius — DISMISSED** [review]

File: server/src/database.js
Comment (bqqbarbhg) [reply server/src/database.js:82]: Could do this, but separate PR

**Review by bqqbarbhg — COMMENTED** [review]
SMASH THAT MF MERGE BUTTON

**Review by joonalillfors — APPROVED** [review]
Ship it

**Pull request merged**
